src/pages/Dashboard/Modals/BorrowRepay/Repay/index.tsx

```diff
@@ -0,0 +1,238 @@
+            label={t('borrowRepayModal.repay.currentlyBorrowing')}
+          >
+            {readableTokenBorrowBalance}
+          </LabeledInlineContent>
+
+          <div css={[styles.getRow({ isLast: true })]}>
+            <TokenTextField
+              name="amount"
+              tokenId={asset.id}
+              css={styles.input}
+              value={values.amount}
+              onChange={amount => setFieldValue('amount', amount, true)}
+              disabled={isRepayLoading}
+              onBlur={handleBlur}
+              rightMaxButton={{
+                label: t('borrowRepayModal.repay.rightMaxButtonLabel'),
+                valueOnClick: limitTokens,
+              }}
+              data-testid="token-text-field"
+              // Only display error state if amount is higher than limit
+              hasError={errors.amount === ErrorCode.HIGHER_THAN_MAX}
+            />
+
+            <Typography
+              component="div"
+              variant="small2"
+              css={[styles.greyLabel, styles.walletBalance]}
+            >
+              <Trans
+                i18nKey="borrowRepayModal.repay.walletBalance"
+                components={{
+                  White: <span css={styles.whiteLabel} />,
+                }}
+                values={{ balance: readableTokenWalletBalance }}
+              />
+            </Typography>
+
+            <div css={styles.selectButtonsContainer}>
+              {[25, 50, 75, 100].map(percentage => (
+                <TertiaryButton
+                  key={`select-button-${percentage}`}
+                  css={styles.selectButton}
+                  onClick={() =>
+                    setFieldValue('amount', getTokenBorrowBalancePercentageTokens(percentage), true)
+                  }
+                >
+                  {formatToReadablePercentage(percentage)}
+                </TertiaryButton>
+              ))}
+            </div>
+          </div>
+
+          <AccountData hypotheticalBorrowAmountTokens={-values.amount} asset={asset} />
+
+          <PrimaryButton
+            type="submit"
+            loading={isRepayLoading}
+            disabled={!isValid || !dirty || isRepayLoading}
+            fullWidth
+          >
+            {dirty && isValid
+              ? t('borrowRepayModal.repay.submitButton')
+              : t('borrowRepayModal.repay.submitButtonDisabled')}
+          </PrimaryButton>
+        </>
+      )}
+    </AmountForm>
+  );
+};
+
+export interface IRepayProps {
+  asset: Asset;
+  onClose: () => void;
+}
+
+const Repay: React.FC<IRepayProps> = ({ asset, onClose }) => {
+  const { t } = useTranslation();
+  const { account } = React.useContext(AuthContext);
+
+  const { mutateAsync: repay, isLoading: isRepayLoading } = useRepayVToken({
+    vTokenId: asset.id as VTokenId,
+  });
+
+  const handleRepay: IRepayFormProps['repay'] = async amountWei => {
+    if (!account?.address) {
+      throw new Error(t('errors.walletNotConnected'));
```

Comment:
We have a specific error for Toasts which includes title and description. In this case we aren't using the description.

I wonder too if it isn't better to make it more generic like a UI error that we catch and respond in the interface that could have other properties in the future

Reply:
Ah good point! Missed that.

Yeah I agree. I renamed to `ToastError` to `UiError` to express that intention, let me know your thoughts.

Reply:
Sounds good to me